### PR TITLE
feat(revenue): per-chain truncation badge on All-time column

### PR DESCRIPTION
**Pull request by chapati23**
## Summary

- When a chain crosses `PROTOCOL_FEE_QUERY_LIMIT` (the silent 1000-row hosted Hasura cap), each pool's All-time total becomes a lower bound. This PR surfaces that with the same `≈` marker the per-row `unpriced` flag uses, but scoped to the **All-time column only** — recent windows (24h/7d/30d) are unaffected because transfers return newest-first.
- New `chainTruncated: boolean` field on `PoolFeeRow`, sourced from `network.fees?.isTruncated`. `approxAnnotation` checks both `row[unpricedField]` and (for the All-time column specifically) `row.chainTruncated`. Per-row unpriced wins when both apply (more informative tooltip).
- 5 component tests added covering all four (truncated, unpriced) combinations.
- Closes the codex P2 follow-up tracked in BACKLOG.md (entry removed in this PR).

## Test plan

- [x] `pnpm --filter ui-dashboard typecheck`
- [x] `pnpm --filter ui-dashboard test -- --run protocol-fees revenue queries`
- [x] `pnpm --filter ui-dashboard lint`
- [ ] Smoke-check on Vercel preview after deploy — capped-chain visual with the new badge.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Low Risk**
> Low risk: changes are isolated to UI annotation logic and add tests; main risk is incorrect window cutoffs/oldest-transfer assumptions causing misleading ≈ badges on the revenue table.
> 
> **Overview**
> Adds **per-window truncation detection** to `RevenueByPoolTable`, showing an `≈` prefix + tooltip when a chain hits the `PROTOCOL_FEE_QUERY_LIMIT` *and* the oldest returned transfer falls within the displayed window, so only affected columns (24h/7d/30d/All-time) are marked as lower bounds.
> 
> Extends row data with `truncated*` flags, updates `approxAnnotation` to prioritize unpriced-token approximation over truncation, and adds component tests covering truncated/unpriced combinations plus per-window cutoff scenarios. Also updates `BACKLOG.md` to mark the truncation leaderboard follow-up as done.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 08bef14d649e83b54494d4afc342d638d5ec1caa. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by chapati23** [comment]
@claude review

**Comment by chapati23** [comment]
cursor review

File: ui-dashboard/src/components/revenue-by-pool-table.tsx [hunk 1]
```diff
@@ -145,6 +153,16 @@ function approxAnnotation(
         "Some transfers from this pool used unpriced/unknown tokens in this window — total is a lower bound.",
     };
   }
+  // Truncation only applies to the All-time column (identified by its unpriced
+  // field). Recent windows (24h/7d/30d) are unaffected because Hasura returns
+  // rows newest-first, so the cap never clips recent history.
+  if (unpricedField === "unpriced" && row.chainTruncated) {
```
[the remark this answers is not in the document]
Comment (chapati23) [reply ui-dashboard/src/components/revenue-by-pool-table.tsx]: Fixed in e59c5c0. Right call — the previous heuristic ("recent windows are unaffected because rows return newest-first") only holds while a chain has fewer than 1000 fee transfers within each displayed window, and that's a fragile invariant to bake into the UI.

`buildRows` now derives a per-window flag from the oldest returned transfer's timestamp:

```ts
const oldest = Number(n.feeTransfers[n.feeTransfers.length - 1].blockTimestamp);
const truncated24h = chainTruncated && oldest > cutoff24h;
// 7d, 30d similarly
const truncatedAll = chainTruncated;
```

Each `FeeColumn` carries its own `truncatedField` symmetric to `unpricedField`, so `approxAnnotation(row, column)` checks both per column with no string-coupling. A capped chain whose oldest returned transfer is, say, 14 days ago now correctly shows `≈` on 30d + All-time but leaves 7d / 24h clean.

File: ui-dashboard/src/components/__tests__/revenue-by-pool-table.test.tsx [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    const cells = renderFeeCells([networkData(transfers, feeSummary(true))]);
+
+    // All-time should be ≈
+    expect(cells.feesAll).toContain("≈");
+    // Recent windows must NOT be ≈ (no unpriced transfers, cap only clips history)
+    expect(cells.fees24h).not.toContain("≈");
+    expect(cells.fees7d).not.toContain("≈");
+    expect(cells.fees30d).not.toContain("≈");
+  });
+});
```
[the remark this answers is not in the document]
Comment (chapati23) [reply ui-dashboard/src/components/__tests__/revenue-by-pool-table.test.tsx:309]: Fixed in e59c5c0. The "recent windows always clean" assertion is gone. Replaced by a series of cases that drive the badge from the oldest returned transfer's timestamp:

- Capped + oldest older than 30d → only All-time `≈`
- Capped + oldest inside 30d (14d ago in the test) → 30d + All-time `≈`, 24h/7d clean
- Capped + oldest inside 7d (3d ago) → 7d + 30d + All-time `≈`, 24h clean
- Capped + oldest inside 24h (1h ago) → every window `≈`
- Not capped → no truncation `≈` regardless of how recent the oldest is

Plus a tooltip-content test that verifies the cap number tracks `PROTOCOL_FEE_QUERY_LIMIT` instead of being hard-coded.

**Comment by chapati23** [comment]
Both findings fixed in e59c5c0:

- **Time/window/bucket: `isTruncated` ≠ recent-windows complete** → Fixed. Per-window flag derived in `buildRows` from `chainTruncated && oldest > windowLowerBound`. Each `FeeColumn` carries a `truncatedField` symmetric to `unpricedField`; `approxAnnotation` checks both per column. A chain that crosses the cap inside 30d now correctly badges 30d + All-time and leaves 24h/7d clean. Extreme case (cap clipped data inside 24h) badges every window.
- **Tests codifying the broken assumption** → Fixed. Replaced the "recent windows always clean" test with five window-boundary cases driven by the oldest returned timestamp, plus a tooltip-content test that locks the cap number to `PROTOCOL_FEE_QUERY_LIMIT.toLocaleString()`.

Side benefit: claude[bot]'s minor finding from the same PR (couple "is All-time?" to a proper boolean instead of `unpricedField === "unpriced"` string equality) is also resolved by the symmetric per-column flag, and the second minor (tooltip text hardcoded the cap number) is fixed by the template-literal change.

**Comment by chapati23** [comment]
@claude — addressed both observations in e59c5c0, plus the cursor blocker that landed at the same time. Summary:

- **`approxAnnotation` couples All-time detection to `unpricedField === "unpriced"`** → Fixed. Each `FeeColumn` now carries an explicit `truncatedField` (`"truncated24h"` / `"truncated7d"` / `"truncated30d"` / `"truncatedAll"`) symmetric to `unpricedField`. `approxAnnotation(row, column)` reads `row[column.truncatedField]` directly with no string-equality on an unrelated field. The signature change naturally addresses the case you raised about a future column being assigned `unpricedField: "unpriced"`.
- **Tooltip hardcodes 1000** → Fixed. Now `\`Chain hit the ${PROTOCOL_FEE_QUERY_LIMIT.toLocaleString()}-row query cap inside this window — total is a lower bound for this chain.\``. Imported `PROTOCOL_FEE_QUERY_LIMIT` from `@/lib/protocol-fees`. Locked by a new test that asserts the rendered cell contains `PROTOCOL_FEE_QUERY_LIMIT.toLocaleString()`.

Bonus: cursor's correctness blocker on this same PR ("`isTruncated` only proves global cap; recent windows can still be lower bounds if the chain crossed 1000 transfers inside a window") landed in the same commit. The badge is now per-window, derived from the oldest returned transfer's timestamp, so a chain whose oldest transfer is 14 days old correctly shows `≈` on 30d + All-time and leaves 7d / 24h clean. Tests rewritten around the window-boundary cases.

Test-file multi-paragraph JSDoc trimmed to a single paragraph so it doesn't violate the no-multi-paragraph-docstring guideline you flagged.
